src/t8_cmesh/t8_cmesh.c

```diff
@@ -3379,3 +3552,262 @@ t8_cmesh_new_hybrid_gate_deformed (sc_MPI_Comm comm)
+  vertices[8] = 2;
+
+  vertices[9] = 1;
+  vertices[10] = 2;
+  vertices[11] = 1;
+
+  vertices[12] = 0;
+  vertices[13] = 2;
+  vertices[14] = 1;
+
+  vertices[15] = 1;
+  vertices[16] = 2;
+  vertices[17] = 2;
+
+  t8_cmesh_set_tree_vertices (cmesh, 3, vertices, 6);
+
+  t8_cmesh_commit (cmesh, comm);
+  return cmesh;
+}
+
+t8_cmesh_t
+t8_cmesh_new_pyramid_cake (sc_MPI_Comm comm, int num_of_pyra)
+{
+  /*num_of_pyra pyras a 5 vertices a 3 coords */
+  /* TODO: This seems too be a lot of memory, can we also get by with only
+     5 * 3 doubles? */
+  int                 i, j;
+  double             *vertices = T8_ALLOC (double, num_of_pyra * 5 * 3);
+  t8_cmesh_t          cmesh;
+  double              degrees = 360. / num_of_pyra;
+  t8_geometry_c      *linear_geom = t8_geometry_linear_new (3);
+
+  if (vertices)
+    T8_ASSERT (num_of_pyra > 2);
+
+  for (i = 0; i < num_of_pyra; i++) {
+    for (j = 0; j < 5; j++) {
+      /*Get the edges at the unit circle */
+      if (j == 4) {
+        vertices[i * 5 * 3 + j * 3] = 0;
+        vertices[i * 5 * 3 + j * 3 + 1] = 0;
+        vertices[i * 5 * 3 + j * 3 + 2] = 0;
+      }
+      else if (j == 1 || j == 3) {
+        vertices[i * 5 * 3 + j * 3] = cos (i * degrees * M_PI / 180);
+        vertices[i * 5 * 3 + j * 3 + 1] = sin (i * degrees * M_PI / 180);
+        vertices[i * 5 * 3 + j * 3 + 2] = (j == 3 ? 0.5 : -0.5);
+      }
+      else if (j == 0 || j == 2) {
+        vertices[i * 5 * 3 + j * 3] =
+          cos ((i * degrees + degrees) * M_PI / 180);
+        vertices[i * 5 * 3 + j * 3 + 1] =
+          sin ((i * degrees + degrees) * M_PI / 180);
+        vertices[i * 5 * 3 + j * 3 + 2] = (j == 2 ? 0.5 : -0.5);
+      }
+    }
+  }
+
+  t8_cmesh_init (&cmesh);
+  t8_cmesh_register_geometry (cmesh, linear_geom);
+
+  for (i = 0; i < num_of_pyra; i++) {
+    t8_cmesh_set_tree_class (cmesh, i, T8_ECLASS_PYRAMID);
+  }
+
+  for (i = 0; i < num_of_pyra; i++) {
+    t8_cmesh_set_join (cmesh, i, (i == (num_of_pyra - 1) ? 0 : i + 1), 0, 1,
+                       0);
+  }
+  for (i = 0; i < num_of_pyra; i++) {
+    t8_cmesh_set_tree_vertices (cmesh, i, vertices + i * 15, 5);
+  }
+  t8_cmesh_commit (cmesh, comm);
+  T8_FREE (vertices);
+
+  return cmesh;
+}
+
+t8_cmesh_t
+t8_cmesh_new_long_brick_pyramid (sc_MPI_Comm comm, int num_cubes)
+{
+  double              vertices_coords[24] = {
+    0, 0, 0,
+    1, 0, 0,
+    0, 1, 0,
+    1, 1, 0,
+    0, 0, 1,
+    1, 0, 1,
+    0, 1, 1,
+    1, 1, 1
+  };
+  t8_topidx_t         vertices[5];
+  double              attr_vertices[15];
+  int                 i, j;
+  t8_cmesh_t          cmesh;
+  t8_geometry_c      *linear_geom = t8_geometry_linear_new (3);
+
+  T8_ASSERT (num_cubes > 0);
+  t8_cmesh_init (&cmesh);
+  t8_cmesh_register_geometry (cmesh, linear_geom);
+
+  for (i = 0; i < num_cubes; i++) {
+    for (j = 0; j < 3; j++) {
+      t8_cmesh_set_tree_class (cmesh, i * 3 + j, T8_ECLASS_PYRAMID);
+    }
+    /*in-cube face connection */
```

Comment:
```suggestion
    /* in-cube face connection */
```
Also comments below.